### PR TITLE
Remove some Emacs < 26, and fix some whitespace/indentation

### PR DESCRIPTION
**Pull request by leungbk**
I'm fine with not merging the whitespace/indentation commit if it'll mess up the git history too much.

### CODE REVIEW

File: lsp-mode.el [hunk 1]
```diff
@@ -5666,7 +5657,7 @@ A reference is highlighted only if it is visible in a window."
        "textDocument/semanticTokens/full")
       (t "textDocument/semanticTokens"))
      `(:textDocument ,(lsp--text-document-identifier)
-       ,@(if region (list :range (lsp--region-to-range (car region) (cdr region))) '()))
+                     ,@(if region (list :range (lsp--region-to-range (car region) (cdr region))) '()))
```
Comment (kiennq) [inline lsp-mode.el:5660]: This line should be changed back, emacs indentation is wrong here

File: lsp-mode.el [hunk 2]
```diff
@@ -5853,12 +5844,11 @@ perform the request synchronously."
...
-  (let ((edits (lsp-request "textDocument/rename"
-                            `(:textDocument ,(lsp--text-document-identifier)
-                                            :position ,(lsp--cur-position)
-                                            :newName ,newname))))
-    (when edits
-      (lsp--apply-workspace-edit edits))))
+  (when-let ((edits (lsp-request "textDocument/rename"
+                                 `(:textDocument ,(lsp--text-document-identifier)
+                                                 :position ,(lsp--cur-position)
+                                                 :newName ,newname))))
```
Comment (kiennq) [inline lsp-mode.el:5850]: these lines should be reverted

File: lsp-mode.el (hunk 1 above)
Comment (leungbk) [reply lsp-mode.el]: I edited the `.dir-locals.el` to use different plist indentation in the `lsp-mode` repo. If it seems fine to you then I'll make the plist changes elsewhere also.

File: lsp-mode.el (hunk 2 above)
Comment (leungbk) [reply lsp-mode.el:5850]: Would you mind explaining? I can't spot any obviously-important difference in the expanded forms.

Comment (kiennq) [reply lsp-mode.el:5850]: Sorry I didn't notice the before change diff.
These should be change so the `:position` is aligned with `:textDocument` and so on

**Comment by leungbk** [comment]
Something like this has happened several times today for me: https://github.com/emacs-lsp/lsp-mode/runs/905172961#step:5:112

I never had this problem before, which is weird.

File: lsp-mode.el [hunk 3]
```diff
@@ -3882,28 +3873,29 @@ Only works when mode is 'tick or 'alive."
   (if (zerop length)
       ;; Adding something only, work from start only
       `(:range ,(lsp--range (lsp--point-to-position start)
-                            (lsp--point-to-position start))
-               :rangeLength 0
-               :text ,(buffer-substring-no-properties start end))
+                 (lsp--point-to-position start))
```
Comment (kiennq) [inline lsp-mode.el:3876]: This seems to be wrongly formatted

File: lsp-mode.el [hunk 4]
```diff
@@ -3882,28 +3873,29 @@ Only works when mode is 'tick or 'alive."
...
 
     (if (eq start end)
         ;; Deleting something only
         (if (lsp--bracketed-change-p start length)
             ;; The before-change value is bracketed, use it
             `(:range ,(lsp--range (lsp--point-to-position start)
-                                  (plist-get lsp--before-change-vals :end-pos))
-                     :rangeLength ,length
-                     :text "")
+                       (plist-get lsp--before-change-vals :end-pos))
```
Comment (kiennq) [inline lsp-mode.el:3885]: same here

File: lsp-mode.el [hunk 5]
```diff
@@ -5332,9 +5324,9 @@ It will show up only if current point has signature help."
                          (propertize
                           lsp-overlay-document-color-char
                           'face `((:foreground ,(format "#%s%s%s"
-                                                        (lsp--number->color red)
-                                                        (lsp--number->color green)
-                                                        (lsp--number->color blue))))
+                                                 (lsp--number->color red)
```
Comment (kiennq) [inline lsp-mode.el:5327]: same here

**Comment by yyoncho** [comment]
Fixing the indentation of plists is really nice idea. FTR we use alists in some methods (e. g. lsp--client-capabilities) due to the fact that plists cannot be formatted nicely.

I checked out the source, and then I applied M-x indent-region over the buffer I saw a lot of changes in it, and some of them seem to be unwanted. Do you see the same thing?

**Comment by leungbk** [comment]
> I checked out the source, and then I applied M-x indent-region over the buffer I saw a lot of changes in it, and some of them seem to be unwanted. Do you see the same thing?

Yeah, this isn't a great solution. It might be easier in the long run to have `lisp-indent-function` as `common-lisp-indent-function` in the `.dir-locals.el` and just deal with the different indentation of CL (or patch the indentation we don't like); at least that will work with `indent-region`.

File: lsp-mode.el (hunk 3 above)
Comment (leungbk) [reply lsp-mode.el:3877]: Should be fixed.

File: lsp-mode.el (hunk 4 above)
Comment (leungbk) [reply lsp-mode.el:3887]: Should be fixed.

File: lsp-mode.el (hunk 5 above)
Comment (leungbk) [reply lsp-mode.el:5337]: Should be fixed.

**Comment by kiennq** [comment]
> Yeah, this isn't a great solution. It might be easier in the long run to have `lisp-indent-function` as `common-lisp-indent-function` in the `.dir-locals.el` and just deal with the different indentation of CL (or patch the indentation we don't like); at least that will work with `indent-region`.

Oh, I've done that (that's why `&allow-other-keys` is indent slightly different with other `&`) and it will become really troublesome soon, we need to patch nearly all macro, especially `dash.el`

**Comment by yyoncho** [comment]
ATM if the user keeps customizing section in his init.el the function evaluation will be inserted in it. Apart from moving some of the definitions that are currently in .dir-locals.el to lsp-mode.el do you know of any other solution to avoid that?

**Comment by leungbk** [comment]
> ATM if the user keeps customizing section in his init.el the function evaluation will be inserted in it. Apart from moving some of the definitions that are currently in .dir-locals.el to lsp-mode.el do you know of any other solution to avoid that?

Sorry, I don't know how to avoid this.

**Pull request merged**